src/Microsoft.Diagnostics.Runtime/src/Common/MemoryReader.cs

```diff
@@ -61,11 +62,11 @@ internal bool TryReadPtr(ulong addr, out ulong value)
             if (_currPageStart <= addr && addr - _currPageStart < (uint)_currPageSize)
             {
                 ulong offset = addr - _currPageStart;
-                fixed (byte* b = &_data[offset])
-                    if (IntPtr.Size == 4)
-                        value = *((uint*)b);
-                    else
-                        value = *((ulong*)b);
+                ref byte b = ref _data[offset];
```

Comment:
```suggestion
                value = Unsafe.As<byte, UIntPtr>(ref _date[offset]).ToUInt64();
```

Maybe using `UIntPtr` here?